File: framework/ios/Example/Example/OpenpathWrapper/Models/UnlockResponse.swift
import Foundation

struct UnlockResponse: Codable, CustomDebugStringConvertible {
    var debugDescription: String {
        let encoder = JSONEncoder()
        return String(data: try! encoder.encode(self), encoding: .utf8)!
    }

    let requestId: Int
    let status: Int
    let intCode: Int?
    let result: String?
    let description: String?
    let tip: String?
    let localizedTip: String?
    let moreInformationLink: String?
    let itemId: Int?
    let itemType: String?
}

File: types/Events/UnlockResponse.ts
/**
 * The data included with the "onUnlockResponse" event.
 *
 * Invoked when {@link unlock} is complete.
 *
 * @event onUnlockResponse
 * @see {@link OpenpathEvent}
 */
export interface UnlockResponse {
  requestId: number;
  status: number;
  intCode?: number;
  result?: string;
  description?: string;
  tip?: string;
  moreInformationLink?: string;
  itemId: number;
  itemType: string;
}
